modules/anonymisedRtdProvider.js

```diff
@@ -42,17 +44,38 @@ export function createRtdProvider(moduleName) {
       return null;
     }
   }
+  /**
+   * Load the Anonymised Marketing Tag script
+   * @param {Object} config
+   */
+  function loadMarketingTag(config) {
+    logMessage(`${SUBMODULE_NAME}RtdProvider: Loading Marketing Tag`);
+    // Check if the script is already loaded
+    if (document.querySelector(`script[src*="${config.params.tagUrl ?? MARKETING_TAG_URL}"]`)) {
+      logMessage(`${SUBMODULE_NAME}RtdProvider: Marketing Tag already loaded`);
+      return;
+    }
+    const tagConfig = config.params?.tagConfig ? {...config.params.tagConfig, idw_client_id: config.params.tagConfig.clientId} : {};
+    delete tagConfig.clientId;
+
+    const tagUrl = config.params.tagUrl ? config.params.tagUrl : `${MARKETING_TAG_URL}?ref=prebid`;
+
+    loadExternalScript(tagUrl, MODULE_TYPE_RTD, SUBMODULE_NAME, () => {
```

Comment:
this loads the tag whether or not tagConfig is defined; change this so tagConfig must be defined for this external fire to occur and make it clear in your docs what the behavior will be if this is defined or not.

Reply:
Great, maybe this is just an edge case then: If config.params.tagConfig is falsy (e.g., undefined), tagConfig simply becomes an empty object rather than causing an error. The call to loadExternalScript then still executes with that empty object, so the marketing tag loads regardless of whether tagConfig is defined

can you do a quick typecheck up front? maybe: 
```
  const clientId = config?.params?.tagConfig?.clientId;
  if (typeof clientId !== 'string' || !clientId.trim()) {
    logWarn(`${SUBMODULE_NAME}RtdProvider: clientId missing or invalid; tag not loaded.`);
    return;
  }
```